SU2_CFD/src/drivers/CMultizoneDriver.cpp

```diff
@@ -680,6 +682,15 @@ bool CMultizoneDriver::Monitor(unsigned long TimeIter){
 
     /*--- Check whether the outer time integration has reached the final time ---*/
     TimeConvergence = GetTimeConvergence();
+
+    if(!cauchyTimeConverged && TimeConvergence && config_container[iZone]->GetTime_Marching() == DT_STEPPING_2ND){
+         // Change flags for 2nd order Time stepping: In case of convergence, this iter and next iter gets written out. then solver stops
+        cauchyTimeConverged = TimeConvergence;
+        TimeConvergence = false;
+    }
+    else if(cauchyTimeConverged){
+        TimeConvergence = cauchyTimeConverged;
+    }
```

Comment:
Please implement the solution in the output so that you do not have to duplicate code.

Reply:
Sorry for the late reply. Had lot to do. I'll take a look

Reply:
Done :)